clients/bmp_client_test.go

```diff
@@ -79,4 +79,224 @@ var _ = Describe("BMP client", func() {
+				End_time:    "fake-end-time-0"}))
+			Expect(tasksResponse.Data[1]).To(Equal(clients.Task{
+				Id:          1,
+				Description: "fake-description-1",
+				Start_time:  "fake-start-time-1",
+				Status:      "fake-status-1",
+				End_time:    "fake-end-time-1"}))
+		})
+
+		It("fails when BMP server /tasks?latest= fails", func() {
+			fakeHttpClient.DoRawHttpRequestError = errors.New("fake-error")
+
+			_, err := bmpClient.Tasks(10)
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("#taskOutput", func() {
+		BeforeEach(func() {
+			fakeHttpClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("..", "bmp", "TaskOutput.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("returns an array of task ouput", func() {
+			taskOutputResponse, err := bmpClient.TaskOutput(10, "event")
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(len(taskOutputResponse.Data)).To(Equal(2))
+			Expect(taskOutputResponse.Data[0]).To(Equal("INFO -- event0"))
+			Expect(taskOutputResponse.Data[1]).To(Equal("ERROR -- event1"))
+		})
+
+		It("fails when BMP server /task/{taskid}/txt/{level} fails", func() {
+			fakeHttpClient.DoRawHttpRequestError = errors.New("fake-error")
+
+			_, err := bmpClient.TaskOutput(10, "event")
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("#updateStatus", func() {
+		BeforeEach(func() {
+			fakeHttpClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("..", "bmp", "UpdateStatus.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("returns an status after updating status", func() {
+			updateStatusResponse, err := bmpClient.UpdateStatus("fake-id", "fake-status")
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(updateStatusResponse.Status).To(Equal("200"))
+		})
+
+		It("fails when BMP server /baremetal/{serverId}/{staus} fails", func() {
+			fakeHttpClient.DoRawHttpRequestError = errors.New("fake-error")
+
+			_, err := bmpClient.UpdateStatus("fake-id", "fake-status")
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("#login", func() {
+		BeforeEach(func() {
+			fakeHttpClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("..", "bmp", "Login.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("returns an status after logining", func() {
+			loginResponse, err := bmpClient.Login("fake-username", "fake-password")
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(loginResponse.Status).To(Equal("200"))
+		})
+
+		It("fails when BMP server /login/{username}/{password} fails", func() {
+			fakeHttpClient.DoRawHttpRequestError = errors.New("fake-error")
+
+			_, err := bmpClient.Login("fake-username", "fake-password")
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("#CreateBaremetal", func() {
+		BeforeEach(func() {
+			fakeHttpClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("..", "bmp", "CreateBaremetal.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		fakeServerSpec := clients.ServerSpec{
```

Comment:
These should be inside the `BeforeEach` not in between it and the `It` statements